tests/core/config/test_data_node_config.py

```diff
@@ -141,7 +133,7 @@ def test_block_datanode_config_update_in_development_mode():
     assert Config.data_nodes[data_node_id].default_path == "foo.p"
     assert Config.data_nodes[data_node_id].storage_type == "pickle"
     assert Config.data_nodes[data_node_id].scope == Scope.SCENARIO
-    assert Config.data_nodes[data_node_id].properties == {"default_path": "foo.p"}
+    assert Config.data_nodes[data_node_id].properties == {"default_data": None, "default_path": "foo.p"}
```

Comment:
First, I don't get why we have this new behavior. Where does the `"default_data": None` comes from?

Second, I believe this is a bug. I am guessing in the python_code_config, default_data has been set to None. We don't want to store a `None` property in the config if the user does not provide it explicitly. At least for two reasons:
1. If there is a default_data provided in the default_data_node_config, it will be overridden by None.
2. If we store None, how to differentiate this case from the case where the user does provide None as default_data

Then, I was surprised to discover this bug in these tests that are not maid for that. 
We should have created dedicated tests that guarantee that we don't store in properties some None attributes that are not explicitly passed in the class constructor.

Reply:
Yes this is because the configure_pickle_data_node has default_data is None.

I'll try to figure out a way to avoid this (and other None default values as well).

Reply:
Hi @jrobinAV, the behavior is actually bug when calling `config_pickle_datanode()`, which mean that None is pass to the default_data as well

Reply:
Resolve by the latest commit